File: DSA/LC/271_encode_and_decode_strings.py
from typing import List


class Solution:

    def encode(self, strs: List[str]) -> str:
        delimiter = '\u001F'
        if len(strs) == 0:
            return "\u001F"
        return delimiter.join(strs)

    def decode(self, s: str) -> List[str]:
        delimiter = '\u001F'
        if s == delimiter:
            return []
        return s.split(delimiter)

File: DSA/LC/271_encode_and_decode_strings.js
function encode(strs) {
  return strs.map((str) => `${str.length}#${str}`).join("");
}
function decode(str) {
  let decodedWords = [];
  let i = 0;
  while (i < str.length) {
    let j = i;
    while (str[j] !== "#") {
      j++;
    }
    let len = parseInt(str.slice(i, j), 10);
    decodedWords.push(str.slice(j + 1, j + 1 + len));
    i = j + 1 + len;
  }
  return decodedWords;
}
export {};
